kerchunk/grib2.py

```diff
@@ -975,3 +1135,257 @@ def build_idx_grib_mapping(
+) -> "pd.DataFrame":
+    """
+    Main method used for building index dataframes from parsed IDX files
+    merged with the correct mapping for the horizon
+
+    Parameters
+    ----------
+
+    run_time : pd.Timestamp
+        the run time timestamp of the idx data
+    mapping : pd.DataFrame
+        the mapping data derived from comparing the idx attributes to the
+        CFGrib attributes for a given horizon
+    idxdf : pd.DataFrame
+        the dataframe of offsets and lengths for each grib message and its
+        attributes derived from an idx file
+    raw_merged : bool
+        Used for debugging to see all the columns in the merge. By default,
+        it returns the index columns with the corrected time values plus
+        the index metadata
+
+    Returns
+    -------
+
+    pd.Dataframe : the index dataframe that will be used to read variable data from the grib file
+    """
+
+    idxdf = idxdf.reset_index().set_index("attrs")
+    mapping = mapping.reset_index().set_index("attrs")
+    mapping.drop(columns="uri", inplace=True)  # Drop the URI column from the mapping
+
+    if not idxdf.index.is_unique:
+        raise ValueError("Parsed idx data must have unique attrs to merge on!")
+
+    if not mapping.index.is_unique:
+        raise ValueError("Mapping data must have unique attrs to merge on!")
+
+    # Merge the offset and length from the idx file with the varname, step and level from the mapping
+
+    result = idxdf.merge(mapping, on="attrs", how="left", suffixes=("", "_mapping"))
+
+    if raw_merged:
+        return result
+    else:
+        # Get the grib_uri column from the idxdf and ignore the uri column from the mapping
+        # We want the offset, length and uri of the index file with the varname, step and level of the mapping
+        selected_results = result.rename(columns=dict(grib_uri="uri"))[
+            [
+                "varname",
+                "typeOfLevel",
+                "stepType",
+                "name",
+                "step",
+                "level",
+                "time",
+                "valid_time",
+                "uri",
+                "offset",
+                "length",
+                "inline_value",
+            ]
+        ]
+    # Drop the inline values from the mapping data
+    selected_results.loc[:, "inline_value"] = None
+    selected_results.loc[:, "time"] = run_time
+    selected_results.loc[:, "valid_time"] = (
+        selected_results.time + selected_results.step
+    )
+    logger.info("Dropping %d nan varnames", selected_results.varname.isna().sum())
+    selected_results = selected_results.loc[~selected_results.varname.isna(), :]
+    return selected_results.reset_index(drop=True)
+
+
+def reinflate_grib_store(
+    axes: list["pd.Index"],
+    aggregation_type: AggregationType,
+    chunk_index: "pd.DataFrame",
+    zarr_ref_store: dict,
+) -> dict:
+    """
+    Given a zarr_store hierarchy, pull out the variables present in the
```

Comment:
There should be a one-line summary at the start of the docstring, since this is what help() and sphynx use as the title describing the method.